packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -908,6 +930,153 @@ class _LargeTitleNavigationBarSliverDelegate
+
+  @override
+  _RenderLargeTitle createRenderObject(BuildContext context) {
+    return _RenderLargeTitle(
+      navBarConstraints: constraints,
+      maxExtent: maxExtent,
+      textDirection: textDirection,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderLargeTitle renderObject) {
+    renderObject
+      ..navBarConstraints = constraints
+      ..maxExtent = maxExtent
+      ..textDirection = textDirection;
+  }
+}
+
+class _RenderLargeTitle extends RenderProxyBox {
+  _RenderLargeTitle({
+    required BoxConstraints navBarConstraints,
+    required double maxExtent,
+    required TextDirection textDirection,
+  }) : _navBarConstraints = navBarConstraints,
+      _maxExtent = maxExtent,
+      _textDirection = textDirection;
+
+  BoxConstraints get navBarConstraints => _navBarConstraints;
+  BoxConstraints _navBarConstraints;
+  set navBarConstraints(BoxConstraints value) {
+    if (_navBarConstraints == value) {
+      return;
+    }
+    _navBarConstraints = value;
+
+    markNeedsLayout();
+  }
+
+  double get maxExtent => _maxExtent;
+  double _maxExtent;
+  set maxExtent(double value) {
+    if (_maxExtent == value) {
+      return;
+    }
+    _maxExtent = value;
+
+    markNeedsLayout();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+
+    markNeedsLayout();
+  }
+
+  Matrix4? _transform;
+
+  Matrix4? get _effectiveTransform {
+    // Maximum scale lets us prevent the large title
+    // from getting clipped when it's width is greater than
+    // the navigation bar's max width constraint.
+    final double maxScale = size.width != 0.0
+      ? clampDouble((navBarConstraints.maxWidth - _kNavBarEdgePadding) / size.width, 1.0, 1.1)
+      : 1.1;
+
+    // This scale is estimated from the settings app in iOS 14.
+    // The large title scales linearly from 1.0 up to 1.15 magnification.
+    // The `navBarConstraints.maxHeight` value is the height of the nav bar,
+    // and `maxExtent` is the default large title height the nav bar snaps back to.
+    // The difference between the two heights is used to scale the title.
+    final double scale = clampDouble(1.0 + (navBarConstraints.maxHeight - maxExtent) / maxExtent * 0.12, 1.0, maxScale);
+
+    final Alignment resolvedAlignment = AlignmentDirectional.bottomStart.resolve(textDirection);
+
+    final Matrix4 resultMatrix = Matrix4.identity();
+
+    Offset? translation;
+    translation = resolvedAlignment.alongSize(size);
+
+    resultMatrix.translate(translation.dx, translation.dy);
+    resultMatrix.scale(scale, scale, 1.0);
+    resultMatrix.translate(-translation.dx, -translation.dy);
+
+    return resultMatrix;
+  }
+
+  @override
+  void performLayout() {
+    child?.layout(constraints, parentUsesSize: true);
+    size = child!.size;
```

Comment:
Also `size` should probably be the scaled by the scale factor too? Otherwise the parent render box is going to cull its hit test box to the pre-scaled size I think?

Reply:
Wouldn't scaling `size` make the assertion fail that the `size` computed by the `computeDryLayout` must be the same one done by `performLayout`?

Reply:
Usually people don't tap on the large title when they're dragging the list view, but I think there're other things that rely on hit-testing too like mouse region.

Reply:
Yeah, you are right, hit-testing is not working properly with current implementation.
I will try to fix that.

Reply:
I got hit tests working by using `RenderShiftedBox` and scaling its size along with paint transform, so the parent is not culling hit test box.
Also, I got rid of the `Align` widget and I could do the same with the `OverflowBox` but this would break some tests that expect to find it in the tree, so I don't know if that's ok.

Reply:
But there is one problem I got stuck with, when switching to RTL text direction something weird is going on with the hit tests.
The hit test box is properly transformed, but the actual hit position seems to be off by some logical pixels (possibly the value of child's parent data offset, since it only applies to the RTL layout)

https://user-images.githubusercontent.com/57679865/193345463-77f53843-43b9-45a1-9431-7dd1fa05357d.mp4

Reply:
Ok, I finally fixed it.
I can't believe I missed such an obvious thing, I forgot to call super method on `applyPaintTransform`, everything is working now.